misc/dist/macos-entitlements-data.xml

```diff
@@ -0,0 +1,18 @@
+<?xml version="1.0" encoding="UTF-8"?>
```

Comment:
misc/dist is specifically for files that support our (now defunct) debian package—let's put this in misc/python/materialize/cli/run/macos-entitlements-data.xml instead.